packages/kit/types/hooks.d.ts

```diff
@@ -1,11 +1,13 @@
-import { IncomingRequest, ParameterizedBody } from './app';
-import { MaybePromise, ResponseHeaders } from './helper';
+import { ParameterizedBody, RawBody } from './app';
+import { MaybePromise, RequestHeaders, ResponseHeaders } from './helper';
 
 export type StrictBody = string | Uint8Array;
 
-export interface ServerRequest<Locals = Record<string, any>, Body = unknown>
-	extends IncomingRequest {
-	origin: string;
+export interface ServerRequest<Locals = Record<string, any>, Body = unknown> {
```

Comment:
this could still extend `IncomingRequest` since the first four fields are shared. It makes the docs look nicer that they're not shared, but I wonder if that's a good enough reason to avoid the inheritance. I really just want the typedocs autogenerated :smile:

Reply:
auto-generated docs are the worst! The types will diverge further when we tackle #70, so I figured we should do what's right for readability here rather than brevity